src/features/filter.ts

```diff
@@ -8,5 +9,29 @@ const initialState = {
 export const filterSlice = createSlice({
   name: 'filter',
   initialState,
-  reducers: {},
+  reducers: {
+    setQuery: (state, action: PayloadAction<string>) => {
+      return {
+        ...state,
+        query: action.payload,
+      };
+    },
+
+    clearQuery: state => {
+      return {
+        ...state,
+        query: '',
+      };
+    },
+
+    setFilter: (state, action: PayloadAction<string>) => {
```

Comment:
The `setFilter` action uses `PayloadAction<string>`, but the `status` field is expected to be of type `Filter`. Ensure that the `PayloadAction` type matches the expected type of the `status` field to prevent runtime errors.